### PR TITLE
fix: Make build:cjs run first and add build action for PR

### PR DESCRIPTION
**Pull request by r4zendev**

### CODE REVIEW

**Review by grzpab — DISMISSED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -12,15 +12,16 @@ jobs:
               with:
                   node-version: 20
             - name: Checkout Repository
-              uses: actions/checkout@v2
+              uses: actions/checkout@v3
```
Comment (grzpab) [inline .github/workflows/build.yml:15]: out of curiosity, why was it needed?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -12,15 +12,16 @@ jobs:
...
                   token: ${{ secrets.GH_TOKEN }}
             - name: Install modules
               uses: pnpm/action-setup@v2
               with:
                   version: 8
-            - name: Install dependencies
-              run: pnpm install
+                  run_install: |
+                      - recursive: true
+                        args: [--frozen-lockfile, --strict-peer-dependencies]
```
Comment (grzpab) [inline .github/workflows/build.yml:24]: I guess the peer deps were the problem?

**Review by grzpab — COMMENTED** [review]
LGTM, let's merge it tomorrow morning and test it

File: .github/workflows/build.yml (hunk 2 above)
Comment (r4zendev) [reply .github/workflows/build.yml:24]: i think recursive was the solution and adding cjs-builder to deps of registry also helped.

File: .github/workflows/build.yml (hunk 1 above)
Comment (r4zendev) [reply .github/workflows/build.yml:15]: for funzies, no specific reason, just to stay with the latest version

**Review by grzpab — APPROVED** [review]

**Pull request merged**